File: Api/Api.Domain/Entities/Unit.cs
using System;
using System.Collections.Generic;

namespace Api.Domain.Entities
{
    /// <summary>
    /// Unit entity
    /// </summary>
    public class Unit : BaseEntity
    {
        /// <summary>
        /// Get or Set Unit Key
        /// </summary>
        public Guid UnitKey { get; set; }

        /// <summary>
        /// Get or Set Unit Name
        /// </summary>
        public string UnitName { get; set; }

        /// <summary>
        /// Get or Set Unit Courses
        /// </summary>
        public IEnumerable<CourseUnit> UnitCourses { get; set; }

        /// <summary>
        /// Get or Set Shifts
        /// </summary>
        public IEnumerable<Shift> Shifts { get; set; }
    }
}

File: App/src/app/models/entities/Unit.ts
import { Guid } from 'guid-typescript';
import { CourseUnit } from './CourseUnit';
import { Shift } from './Shift';

export interface Unit {
  scheduleKey: Guid,
  scheduleVersion: number,
  unitKey: Guid,
  unitName: string,
  unitCourses: CourseUnit[],
  shifts: Shift[],
}

